js/src/tooltip.js

```diff
@@ -646,7 +653,7 @@ class Tooltip extends BaseComponent {
     if (event) {
       context._activeTrigger[
         event.type === 'focusout' ? TRIGGER_FOCUS : TRIGGER_HOVER
-      ] = false
+      ] = context._element.contains(event.relatedTarget)
```

Comment:
This change is not required IMO :thinking:
Since the active trigger should be false when the tooltip is about to be hidden.

Reply:
@rohit2sharma95 I think this is neccessary to make it work with SVG and nested elements that cause mouseover to trigger

Reply:
IMO, now this should be handled by:
```js
if (this._isWithActiveTrigger()) {
  return
}
```

Reply:
Mh maybe you're correct then. Another reason could be that this also prevents `hide.bs.tooltip` event from getting triggered but now idea if that been the reason.

Reply:
I agree with Rohit. 
```js

if (this._isWithActiveTrigger()) {
  return
}
```

seems engough

Reply:
I'm not sure either why exactly it was decided to do it that way but I tried to adapt the behaviour to their native counterparts on the `EventHandler` module.
During my tests on the tooltip page tho I realized there is a problem with the tooltip plugin itself so I rather focused on this problem instead.
Like you, I found the reason behind the tooltip from jumping around to be within the initialization of popper. But I didn't realize the problem with the transition handlers until later. I partly did some clean-up (I started by uncommenting `_hoverState` as someone pointed out to be a problem and eventually got rid of it) and after reading and understanding the code better I eventually realized where the problem comes from but I still had to come up with a proper solution for this.
During testing I also realized the same problem with the element loosing the hover state with the tooltip showing up so I got here to ask if this is something to be fixed as well. Thats when I saw your pull request and well looks like finishing the `EventHandler` would have been a better time investment.
Also I found another problem with the `noop` handlers for the IOS fix - they are not getting removed properly but no idea if this will have an impact on performance.

Reply:
I've [added a test in this commit](https://github.com/twbs/bootstrap/pull/33289/commits/122a4af76c83ff9c60f616cb69dec9e0d146ce62) which show why it important this change is important.